src/Spectre.Service/Loaders/DatasetLoader.cs

```diff
@@ -0,0 +1,137 @@
+﻿/*
+ * DatasetLoader.cs
+ * Class loading datasets from either local root or remote root.
+ *
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Ninject;
+using Spectre.Data.Datasets;
+using Spectre.Service.Configuration;
+
+namespace Spectre.Service.Loaders
+{
+    /// <summary>
+    /// Class for loading datasets from specified directories.
+    /// </summary>
+    public class DatasetLoader
+    {
+        #region Fields
+
+        /// <summary>
+        /// Root for local directory.
+        /// </summary>
+        private readonly string _localRoot;
+
+        /// <summary>
+        /// Root for remote directory.
+        /// </summary>
+        private readonly string _remoteRoot;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetLoader"/> class.
+        /// </summary>
+        /// <param name="dataRootConfig">Validated configuration of needed directories.</param>
+        public DatasetLoader(DataRootConfig dataRootConfig)
+        {
+            _localRoot = dataRootConfig.LocalPath;
+            _remoteRoot = dataRootConfig.RemotePath;
+        }
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        [Inject]
+        public IFileSystem FileSystem { private get; set; }
+
+        #endregion
+
+        #region Loaders
+
+        /// <summary>
+        /// Returns dataset basing directly on the name of the file.
+        /// </summary>
+        /// <param name="name">File name.</param>
+        /// <returns>Found dataset.</returns>
+        /// <exception cref="ArgumentException">Throws when the file is not found both locally and remotely.</exception>
+        public IDataset GetFromName(string name)
+        {
+            string fullPathLocal = _localRoot + "\\" + name;
+
+            if (!FileSystem.File.Exists(fullPathLocal))
+            {
+                string fullPathRemote = _remoteRoot + "\\" + name;
+                if (!FileSystem.File.Exists(fullPathRemote))
+                {
+                    throw new ArgumentException("Dataset \"" + name + "\" not found neither locally nor remotely.");
```

Comment:
Use nameof operator for throwing ArgumentException - https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/nameof .

Reply:
Try using string interpolation for evaluating string expressions. It condenses the code and calls string format underneath, which makes it more localization safe than just adding strings.